rtloader/three/three.cpp

```diff
@@ -50,18 +57,36 @@ Three::~Three()
 
 void Three::initPythonHome(const char *pythonHome)
 {
+    // Py_SetPythonHome stores a pointer to the string we pass to it, so we must keep it in memory
     wchar_t *oldPythonHome = _pythonHome;
     if (pythonHome == NULL || strlen(pythonHome) == 0) {
         _pythonHome = Py_DecodeLocale(_defaultPythonHome, NULL);
     } else {
         _pythonHome = Py_DecodeLocale(pythonHome, NULL);
     }
 
-    // Py_SetPythonHome stores a pointer to the string we pass to it, so we must keep it in memory
     Py_SetPythonHome(_pythonHome);
     PyMem_RawFree((void *)oldPythonHome);
 }
 
+void Three::initPythonExe(const char *python_exe)
+{
+    // Py_SetProgramName stores a pointer to the string we pass to it, so we must keep it in memory
+    wchar_t *oldPythonExe = _pythonExe;
+    _pythonExe = Py_DecodeLocale(python_exe, NULL);
+
+    Py_SetProgramName(_pythonExe);
+
+    // HACK: This extra internal API invocation is due to the workaround for an upstream bug on
```

Comment:
If it's just necessary on windows can we put that line within a preprocessor block?
```
#if defined(_WIN32)
  _Py_SetProgramFullPath(_pythonExe);
#endif
```
I believe this would be accurate but you may want to check with someone more current and knowledgeable on all-things-windows than me 😊

Reply:
I feel a bit conflicted on this one. While we can easily do what you are saying, there are a few upsides to having it without a guard `define` that made me choose otherwise:
- Consistent codebase for all platforms
- It's an effective noop on other platforms afaik since the method above it should be doing the same exact thing. 
- Tests pass fine as-is on all used platforms.
- Problems can be found quicker with building the code as it won't just fail on Windows.
- This function is deprecated w/ 3.11+ so when we start seeing warnings for that we will have it on all the platforms as a good reminder that we should move to the new way for Py configuration.

Overall, the benefits of not platform-guarding that call to me seemed to outweigh the downsides. I'd love your thoughts on it though.

Reply:
OK, I'm fine with this.